Source/WebKit/GPUProcess/graphics/ScopedRenderingResourcesRequest.h

```diff
@@ -82,4 +83,9 @@ inline bool hasOutstandingRenderingResourceUsage()
     return ScopedRenderingResourcesRequest::s_requests;
 }
 
+class ResourceCounter : public ThreadSafeRefCounted<ResourceCounter> {
```

Comment:
Using the refCount as your thread safe counter forces you to heap-allocate these, and that seems wasteful for something that's tiny.

Reply:
On the other hand, there will only be as most `1 + number_of_web_processes` of these objects, so while the overhead per object is large, the total absolute additional memory / heap usage isn't great.

Reply:
Alternative would be to store them as `std::atomic`s and always pointer chase through the `RemoteRenderingBackend` -> `RemoteSharedResourceCache` when accessing them.